nautobot/core/ui/object_detail.py

```diff
@@ -378,25 +385,132 @@ def __init__(
+                except NoReverseMatch:
+                    logger.warning("add route for `body_content_table_model` not found")
+
+        elif self.add_button_route is not None:
+            if request.user.has_perms(self.add_permissions or []):
+                add_route = reverse(self.add_button_route)
+                body_content_table_add_url = f"{add_route}?{related_field_name}={obj.pk}&return_url={return_url}"
+
+        return body_content_table_add_url
+
     def get_extra_context(self, context):
-        """Take the context value under `self.table_key` and add it to the context under key `"body_content_table"`."""
-        return {"body_content_table": context[self.table_key]}
+        """Add additional context for rendering the table panel.
+
+        This method processes the table data, configures pagination, and generates URLs
+        for listing and adding objects. It also handles field inclusion/exclusion and
+        displays the appropriate table title if provided.
+        """
+        body_content_table = context[self.table_key]
+        request = context["request"]
+
+        if self.exclude_fields or self.include_fields:
+            for column in body_content_table.columns:
+                if (self.exclude_fields and column.name in self.exclude_fields) or (
+                    self.include_fields and column.name not in self.include_fields
+                ):
+                    body_content_table.columns.hide(column.name)
+                else:
+                    body_content_table.columns.show(column.name)
+
+        per_page = self.max_display_count if self.max_display_count is not None else get_paginate_count(request)
+        paginate = {"paginator_class": EnhancedPaginator, "per_page": per_page}
+        RequestConfig(request, paginate).configure(body_content_table)
+        more_queryset_count = max(body_content_table.data.data.count() - per_page, 0)
+
+        obj = context["obj"]
+        body_content_table_model = body_content_table.Meta.model
+        related_field_name = self.related_field_name or obj._meta.model_name
+
+        try:
+            list_route = reverse(get_route_for_model(body_content_table_model, "list"))
+            body_content_table_list_url = f"{list_route}?{related_field_name}={obj.pk}"
+        except NoReverseMatch:
+            body_content_table_list_url = None
+
+        body_content_table_add_url = self._get_table_add_url(context)
+        body_content_table_verbose_name_plural = self.table_title or body_content_table_model._meta.verbose_name_plural
+
+        return {
+            "body_content_table": body_content_table,
+            "body_content_table_add_url": body_content_table_add_url,
+            "body_content_table_list_url": body_content_table_list_url,
+            "body_content_table_verbose_name": body_content_table_model._meta.verbose_name,
+            "body_content_table_verbose_name_plural": bettertitle(body_content_table_verbose_name_plural),
```

Comment:
Feels self-inconsistent that `verbose_name` is as-is but `verbose_name_plural` is title-cased. Let's remove the bettertitle here and do that in the template as needed instead.